scripts/utils/llvm-gcov.bat

```diff
@@ -0,0 +1,5 @@
+:: Copyright (c) 2022 Google LLC
```

Comment:
sorry, but this was not what I intended with the comment here:
https://github.com/zephyrproject-rtos/zephyr/pull/52005/files#r1020146144

Adding `.bat` (and `.sh`) files will set an undesired precedence of having platform specific scripts.

A major reason for using CMake and python is to avoid such OS specific scripts so it would much be preferred to keep this in Python.

If you read the guide on llvm-cov it states:
https://www.llvm.org/docs/CommandGuide/llvm-cov.html#description
> If the program is invoked with a base name of gcov, it will behave as if the llvm-cov gcov command were called. 

This means no wrapper script is needed, and we can just create a `gcov` link pointing to the `llvm-cov` executable.
That link can be placed in the twister-out folder.

This is easily done in this file:
https://github.com/zephyrproject-rtos/zephyr/blob/94458f88b98abf33a08cc849a620f813d82afe0b/scripts/pylib/twister/twisterlib/coverage.py#L244-L246


Adjust code to something similar to this:

```diff
+        if use_system_gcov:
+            options.gcov_tool = "gcov"
+        elif os.environ.get("ZEPHYR_TOOLCHAIN_VARIANT") == "llvm":
+            llvm_path = os.environ.get("LLVM_TOOLCHAIN_PATH")
+            if llvm_path is not None:
+                llvm_path = os.path.join(llvm_path, "bin")
+            llvm_cov = shutil.which("llvm-cov", path=llvm_path)
+            llvm_cov_ext = pathlib.Path(llvm_cov).suffix
+            gcov_lnk = os.path.join(options.outdir, f"gcov{llvm_cov_ext}")
+            try:
+                os.symlink(llvm_cov, gcov_lnk)
+            except OSError:
+                shutil.copy(llvm_cov, gcov_lnk)
+            options.gcov_tool = gcov_lnk
+        elif "ZEPHYR_SDK_INSTALL_DIR" not in os.environ:
```
The fallback to the file copy is to support windows environments where users are not allowed to create symlinks.
Depending on developer / administrator rights users may be allowed to create symlinks, but we are not guaranteed.

Note, remember to import pathlib, and shutil in top of file, as well as adjusting this condition:
https://github.com/zephyrproject-rtos/zephyr/blob/94458f88b98abf33a08cc849a620f813d82afe0b/scripts/pylib/twister/twisterlib/coverage.py#L241-L242

Doing this instead will also simplify things for users, as all they need to do is to set `ZEPHYR_TOOLCHAIN_VARIANT=llvm` which they have already done.

Simplified approach with above changes and the result:
```
$ echo $ZEPHYR_TOOLCHAIN_VARIANT
llvm
$ ./scripts/twister --coverage -p unit_testing --coverage-tool lcov -T tests/unit/crc/ -v -v
...
DEBUG   - Select lcov as the coverage tool...
DEBUG   - Working on /projects/github/ncs/zephyr/twister-out/unit_testing/tests/unit/crc/utilities.crc/handler.log
DEBUG   - Generating gcda files
DEBUG   - Gcov data captured: /projects/github/ncs/zephyr/twister-out/unit_testing/tests/unit/crc/utilities.crc/handler.log
DEBUG   - Running lcov --gcov-tool /projects/github/ncs/zephyr/twister-out/gcov --capture --directory /projects/github/ncs/zephyr/twister-out --rc lcov_branch_coverage=1 --output-file /projects/github/ncs/zephyr/twister-out/coverage.info...
geninfo: WARNING: GCOV did not produce any data for /projects/github/ncs/zephyr/twister-out/unit_testing/tests/unit/crc/utilities.crc/CMakeFiles/testbinary.dir/projects/github/ncs/zephyr/subsys/testsuite/ztest/src/ztest_rules.c.gcda
INFO    - HTML report generated: /projects/github/ncs/zephyr/twister-out/coverage/index.html
...
```

Reply:
I was going to look into updating twister to make it easier for us in another PR. The intent of the script was just to make it easy to replicate what I did to test. I'll just remove that part from this PR and do it in a separate PR.